### PR TITLE
No-Jira: serialize cachekeys used to remove non-word characters

### PR DESCRIPTION
**Pull request by jebentier**
NoJira PR to fix a bug in the `cacheStore` and `cacheRestore` functions that were causing builds to crash when having a `/` in a cache key.

### CODE REVIEW

File: vars/cacheStore.groovy [hunk 1]
```diff
@@ -16,11 +16,13 @@ void call(String s3Bucket, ArrayList<String> cacheKeys, ArrayList<String> itemsT
 
   echo 'Pushing cache to AWS'
   cacheKeys.each { cacheKey ->
+    String serializedCacheKey = cacheKey.replaceAll("\\W", "")
```
Comment (alecjacobs5401) [inline vars/cacheStore.groovy]: Is there a way to extract this into a `serialize` function that we can use both here and in `cacheRestore`? This way we can keep them aligned and have a new helper function if we need serialization elsewhere

**Review by alecjacobs5401 — COMMENTED** [review]
@jebentier Just one question really about DRYing up the serialization

File: vars/cacheStore.groovy (hunk 1 above)
Comment (jebentier) [reply vars/cacheStore.groovy]: done

**Review by alecjacobs5401 — APPROVED** [review]
Sweet 👍 

Thanks for tackling this so quickly!

**Pull request merged**